dlme_airflow/models/collection.py

```diff
@@ -5,13 +5,18 @@ class Collection(object):
     def __init__(self, provider, collection):
         self.name = collection
         self.provider = provider
-        self.catalog = catalog_for_provider(f"{provider.name}.{self.name}")
 
-    def label(self):
-        return f"{self.provider.name}_{self.name}"
+    def catalog(self):
```

Comment:
Something to consider, you could annotate with `@property` decorator here if you want to make sure that`catalog` is only every accessed read-only as a property (not a method).

```python
@property 
def label(self):
    return f"{self.provider.name}_{self.name}"
```

Would that still get at the spirit of what you were trying to do?